### PR TITLE
LiBJS: Accept calendar names case-insensitively

### PR DESCRIPTION
**Pull request by yedayak**
This is a normative change in the Temporal spec, one tick in #15525 

See tc39/proposal-temporal@03101c6

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp [hunk 1]
```diff
@@ -37,9 +37,11 @@ bool is_builtin_calendar(String const& identifier)
     // 1. Let calendars be AvailableCalendars().
     auto calendars = available_calendars();
 
-    // 2. If calendars contains id, return true.
-    if (calendars.contains_slow(identifier))
-        return true;
+    // 2. If calendars contains the ASCII-lowercase of id, return true.
+    for (auto it = calendars.begin(); it != calendars.end(); ++it) {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp:41]: Use range-based for

**Review by linusg — CHANGES_REQUESTED** [review]
LiBJS -> LibJS

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp (hunk 1 above)
Comment (yedayak) [reply Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp]: Done

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp [hunk 2]
```diff
@@ -37,9 +37,10 @@ bool is_builtin_calendar(String const& identifier)
     // 1. Let calendars be AvailableCalendars().
     auto calendars = available_calendars();
 
-    // 2. If calendars contains id, return true.
-    if (calendars.contains_slow(identifier))
-        return true;
+    // 2. If calendars contains the ASCII-lowercase of id, return true.
+    for (auto calendar : calendars)
+        if (calendar.equals_ignoring_case(identifier))
+            return true;
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp:43]: Please do use brackets for multiline things, even if they contain single line ifs
```cpp
for (auto calendar : calendars) {
	if (calendar.equals_ignoring_case(identifier))
    	return true;
}
```

**Review by davidot — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp (hunk 2 above)
Comment (yedayak) [reply Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp]: Makes sense, done

**Review by davidot — APPROVED** [review]
LGTM!

**Pull request merged**